spec/requests/users_spec.rb

```diff
@@ -615,19 +615,24 @@
   end
 
   context "Revoking Sessions" do
+    before do
+      # Keep this outside of the loop below with the `let` providing the value
+      # for the stub, otherwise it will not work as expected
+      stub_settings_merge(:server => {:session_store => session_store_value})
+      TokenStore.token_caches.clear
+    end
+
+    after { TokenStore.token_caches.clear }
+
     %w[cache sql memory].each do |session_store|
-      before do
-        ::Settings.server.session_store = session_store
-      end
+      let(:session_store_value) { session_store }
```

Comment:
I guess this doesn't clobber the next one in the loop?  Still seems weird.  I wonder if it's just safer to wrap the entire thing in a context.

Reply:
pretty sure the contexts are not kept across tests.
There were gems that reused contexts across tests to speed things up, but I'm pretty sure that stopped over 5 years go.

Reply:
I'm not sure what you are saying - I'm suggesting:

%w[cache sql memory].each do |session_store|
  context "with #{session_store}" do
    let(:session_store_value) { session_store }

    it "revokes all own sessions of authenticated user with #{session_store}" do
...
```

Reply:
@Fryguy Hmm... I didn't catch what you were saying before, but possibly using the `context` instead of the `let` would solve the issue with the `before` loop.  The internals of `RSpec` are a bit complex, to say the least, but I think if we created a new context and put the `before` inside that, it would only create one `before` for that single context`.

Unsure, but I can try that out later today.  Not sure it is worth it now that we have a working solution, but I will see what comes of that.